src/app/main.cpp

```diff
@@ -896,12 +896,22 @@ int main( int argc, char *argv[] )
   // Set up the QgsSettings Global Settings:
   // - use the path specified with --globalsettingsfile path,
   // - use the environment if not found
+  // - use the AppDataLocation ($HOME/.local/share/QGIS/QGIS3 on Linux, roaming path on Windows)
   // - use a default location as a fallback
   if ( globalsettingsfile.isEmpty() )
   {
     globalsettingsfile = getenv( "QGIS_GLOBAL_SETTINGS_FILE" );
   }
 
+  if ( globalsettingsfile.isEmpty() )
+  {
+    QStringList startupPaths = QStandardPaths::locateAll( QStandardPaths::AppDataLocation, QStringLiteral( "qgis_global_settings.ini" ) );
+    if ( !startupPaths.isEmpty() )
+    {
+      globalsettingsfile = startupPaths.at( 0 );
```

Comment:
Is it possible for the user to see which file has been picked up already?
How about adding a log (info) message here for system debug purpose?

Reply:
This already handled below in the code. If global settings file found a message will be added to QGIS Message Log in the "QGIS" tab, see https://github.com/qgis/QGIS/blob/master/src/app/main.cpp#L922.